engine/src/main/java/org/terasology/rendering/dag/RenderTaskListGenerator.java

```diff
@@ -114,8 +114,8 @@ private void generateTaskList() {
     private void generateIntermediateList(List<Node> orderedNodes) {
         intermediateList.clear();
         for (Node node : orderedNodes) {
+            node.setTaskListGenerator(this);
```

Comment:
Hmmm... I don't think nodes should have access to the task list. StateChanges are the interface between nodes and the task list. Why are we doing this?